### PR TITLE
LIVY-224. Datanucleus jars are not detected for spark 2.x

### PR DESCRIPTION
**Pull request by zjffdu**
This PR is to detect datanucleus jars for spark 2.x where datanucleus is in different locations compared to spark 1.x.

Manually tested.

### CODE REVIEW

**Comment by zjffdu** [comment]
@jerryshao @tc0312 @vanzin  Please help review.

**Comment by jerryshao** [comment]
LGTM.

File: server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala [hunk 1]
```diff
@@ -342,18 +342,29 @@ class InteractiveSession(
...
+            new File(sparkHome, "lib")
+          } else {
+            new File(sparkHome, "lib_managed/jars")
+          }
+        case 2 =>
+          if (new File(sparkHome, "RELEASE").isFile) {
+            new File(sparkHome, "jars")
+          } else if (new File(sparkHome, "assembly/target/scala-2.11/jars").isDirectory) {
+            new File(sparkHome, "assembly/target/scala-2.11/jars")
+          } else {
```
Comment (tc0312) [inline server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala]: Can we do a check here?
} else if (new File(sparkHome, "assembly/target/scala-2.10/jars").isDirectory) {

If all if clauses are false, can we print a warning?

Comment (zjffdu) [inline server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala]: warning message will be printed here if datanucleus jars are not found, https://github.com/zjffdu/livy/blob/LIVY-224/server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala#L375

**Pull request merged**